### PR TITLE
Serilog vendoring

### PR DESCRIPTION
**Pull request by colin-higgins**
Changes proposed in this pull request:
 - [x] Tool to maintain vendored libraries
 - [x] Vendor serilog
 - [x] Update third party licenses file
 - [x] ~Strong sign and reference vendors project~
 - [x] Write managed library logs to file
 - [x] Save commit info with vendored code
 - [x] ~Use nuspec file to deploy vendor library with Datadog.Trace~
 - [x] ~Update WIX files to include new dependencies~
 - [x] Test MSI
 - [x] ~Update CI to build nuget package based on NuSpec~
 - [x] Test nuget

@DataDog/apm-dotnet

### CODE REVIEW

File: tools/UpdateVendors/Program.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        private static readonly string DownloadDirectory = Path.Combine(CurrentDirectory, "downloads");
+        private static string _vendorProjectDirectory = Path.Combine(CurrentDirectory, "downloads");
+
+        public static void Main(string[] args)
+        {
+            InitializeCleanDirectory(DownloadDirectory);
+            var solutionDirectory = EnvironmentHelper.GetSolutionDirectory();
+            _vendorProjectDirectory = Path.Combine(solutionDirectory, "src", "vendors", "Datadog.Trace.Vendoring");
+
+            UpdateVendor(
```
Comment (colin-higgins) [inline tools/UpdateVendors/Program.cs]: We could theoretically use this pattern for any library we decide to vendor in the future.
Any time we want to upgrade, run this.

File: src/Datadog.Trace/Span.cs [hunk 2]
```diff
@@ -24,6 +25,12 @@ internal Span(SpanContext context, DateTimeOffset? start)
             Context = context;
             ServiceName = context.ServiceName;
             StartTime = start ?? Context.TraceContext.UtcNow;
+
+            Log.Debug(
```
Comment (zacharycmontoya) [inline src/Datadog.Trace/Span.cs]: I was actually about to comment here that I wanted to add logging about the traceId and spanId. Regardless of the user's LOGS_INJECTION_ENABLED setting, I was thinking we should always be adding the traceId and spanId with our own managed logging. Do you think we should use the event subscriber approach or do you think this beginning/end logging is sufficient?

Comment (colin-higgins) [reply src/Datadog.Trace/Span.cs]: I think beginning and end logging is sufficient. I don't want to add TOO much to the context, as that'll be logged with everything. Maybe if we start with this and decide it's not enough we can rethink it?

Comment (zacharycmontoya) [reply src/Datadog.Trace/Span.cs]: Sounds good to me, this is already going to be great information to have!

File: src/Datadog.Trace/Span.cs [hunk 3]
```diff
@@ -264,6 +270,17 @@ public void Finish(DateTimeOffset finishTimestamp)
             if (shouldCloseSpan)
             {
                 Context.TraceContext.CloseSpan(this);
+                if (Log.IsEnabled(LogEventLevel.Debug))
```
Comment (zacharycmontoya) [inline src/Datadog.Trace/Span.cs]: Is there a reason this logging statement is hidden under the `Log.IsEnabled` check whereas the span start isn't?

Comment (colin-higgins) [reply src/Datadog.Trace/Span.cs]: To prevent the interpolation. Happy to refactor it otherwise.

We have a bunch of spots in c++ that do this on that note.

File: src/Datadog.Trace/Span.cs [hunk 4]
```diff
@@ -264,6 +270,17 @@ public void Finish(DateTimeOffset finishTimestamp)
...
+                {
+                    var metadata =
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Span.cs]: Should we try to keep all that data in `metadata` structured instead of baking it into a single string? Like you are doing below with `"Span closed: [s_id: {0}, p_id: {1}, t_id: {2}] for {3}"`.

Comment (colin-higgins) [reply src/Datadog.Trace/Span.cs]: I got really sick of index-based interpolation, but I guess so.
it could be useful if we change these all to named properties if we decide to use serilog for stability metrics. I don't see the benefit otherwise.

Comment (lucaspimentel) [reply src/Datadog.Trace/Span.cs]: I think we can still use string interpolation while keeping the original separate values. See [`FormattableString`](https://docs.microsoft.com/en-us/dotnet/api/system.formattablestring).

Comment (colin-higgins) [reply src/Datadog.Trace/Span.cs]: Sorry, I don't understand. Also FormattableString isn't present until 4.6, so it would need a preprocessor directive. I can add these as indexes though, just seemed less readable to me.

File: src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  <PropertyGroup Condition="$(TargetFramework.StartsWith('net4'))">
+    <DefineConstants>$(DefineConstants);ATOMIC_APPEND;HRESULTS</DefineConstants>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
+    <DefineConstants>$(DefineConstants);OS_MUTEX</DefineConstants>
+  </PropertyGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
+    <PackageReference Include="System.IO" Version="4.3.0" />
```
Comment (lucaspimentel) [inline src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj]: Seems weird that we would need to add all of these. For example, `System.IO.FileSystem` has a dependency on `System.IO` and `System.IO.FileSystem.Primitives`, so we don't need to specify those. The new `<PackageReference>` system takes care of transient dependencies. You longer need to list them all like in legacy `packages.config` files.

File: src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
+    <DefineConstants>$(DefineConstants);ASYNCLOCAL;HASHTABLE</DefineConstants>
+  </PropertyGroup>
+
+  <ItemGroup Condition="$(TargetFramework.StartsWith('net4'))">
+    <Reference Include="System" />
+    <Reference Include="System.Core" />
+    <Reference Include="Microsoft.CSharp" />
```
Comment (lucaspimentel) [inline src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj]: nit: we wouldn't need this if we drop `<DisableImplicitFrameworkReferences>` above, which I though we had already agreed on.

File: src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj [hunk 7]
```diff
@@ -0,0 +1,61 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <RootNamespace>Datadog.Trace.Vendoring</RootNamespace>
+    <TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">net45;net46;netstandard2.0</TargetFrameworks>
```
Comment (lucaspimentel) [inline src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj]: This should probably be `net461` instead of `net46` to match `Datadog.Trace` and its NuGet package.
```suggestion
    <TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">net45;net461;netstandard2.0</TargetFrameworks>
```

File: src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netstandard2.0</TargetFrameworks>
+    <GenerateDocumentationFile>false</GenerateDocumentationFile>
+    <LangVersion>7.3</LangVersion>
```
Comment (lucaspimentel) [inline src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj]: Should we just use `<LangVersion>latest</LangVersion>` going forwards? I thought about removing it entirely but the default is `7.3` when targeting anything but `netcoreapp3.0`, which defaults to `8.0`.

[reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version)

File: src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+    <!-- Strong name signature -->
+    <SignAssembly>true</SignAssembly>
+    <AssemblyOriginatorKeyFile>..\..\..\Datadog.Trace.snk</AssemblyOriginatorKeyFile>
+    <PublicSign Condition=" '$(OS)' != 'Windows_NT' ">true</PublicSign>
+    <DelaySign>false</DelaySign>
```
Comment (lucaspimentel) [inline src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj]: I don't think we need this since it's inherited from the root `Directory.Build.props`.

File: src/vendors/Datadog.Trace.Vendoring/DatadogLogging.cs [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+using Datadog.Trace.Vendoring.Serilog.Sinks.File;
+
+namespace Datadog.Trace.Vendoring
+{
+    /// <summary>
+    /// Class for Datadog managed file logging.
+    /// </summary>
+    public static class DatadogLogging
+    {
+        private const string WindowsDefaultDirectory = "C:\\ProgramData\\Datadog .NET Tracer\\logs\\";
```
Comment (lucaspimentel) [inline src/vendors/Datadog.Trace.Vendoring/DatadogLogging.cs]: We should use `System.Environment.GetFolderPath(System.Environment.SpecialFolder.CommonApplicationData)` to determine the path and `Path.Combine()` to combine them.
```suggestion
        private const string WindowsDefaultDirectory = Path.Combine(Environment.GetFolderPath(System.Environment.SpecialFolder.CommonApplicationData), @"Datadog .NET Tracer", "logs");
```

File: src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj (hunk 5 above)
Comment (colin-higgins) [reply src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj]: This is what was in the serilog proj files, it's basically a copy paste.

File: src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj (hunk 6 above)
Comment (colin-higgins) [reply src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj]: Same as above, I'm only taking what's in the serilog proj files. I figure we should do everything they do.

File: src/vendors/Datadog.Trace.Vendoring/DatadogLogging.cs (hunk 10 above)
Comment (colin-higgins) [reply src/vendors/Datadog.Trace.Vendoring/DatadogLogging.cs]: But then it can't be const! :P
Yeah, sounds good.

File: src/vendors/Datadog.Trace.Vendoring/DatadogLogging.cs [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+        private const string NixDefaultDirectory = "/var/log/datadog/";
+
+        private static readonly long? MaxLogFileSize = 10 * 1024 * 1024;
```
Comment (lucaspimentel) [inline src/vendors/Datadog.Trace.Vendoring/DatadogLogging.cs]: Looks like this doesn't need to be nullable, so it can also be `const`.
```suggestion
        private const long MaxLogFileSize = 10 * 1024 * 1024;
```

File: src/Datadog.Trace/Datadog.Trace.nuspec [hunk 12]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0"?>
+<package  xmlns="http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd">
+  <metadata>    
+    <id>Datadog.Trace</id>
+    <version>1.7.0</version>
+    <title>Datadog APM</title>
+    <authors>lucas.pimentel.datadog;colinhigginsdatadog;zachmontoyadd</authors>
+    <owners>Datadog</owners>
+    <licenseUrl>https://github.com/DataDog/dd-trace-dotnet/blob/master/LICENSE</licenseUrl>
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Datadog.Trace.nuspec]: `licenseUrl` is deprecated. See [`src/Directory.Build.props`](https://github.com/DataDog/dd-trace-dotnet/blob/master/src/Directory.Build.props).
```suggestion
    <license>Apache-2.0</license>
```

File: src/Datadog.Trace/Datadog.Trace.nuspec [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+    <projectUrl>https://docs.datadoghq.com/tracing/setup/dotnet/</projectUrl>
+    <iconUrl>https://imgix.datadoghq.com/img/favicons/favicon-32x32.png</iconUrl>
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Datadog.Trace.nuspec]: `iconUrl` is also deprecated, but you can just use this 64x64 icon for now. We should fix it here and in [`src/Directory.Build.props`](https://github.com/DataDog/dd-trace-dotnet/blob/master/src/Directory.Build.props) at some point.
```suggestion
    <iconUrl>https://github.com/DataDog/dd-trace-dotnet/raw/master/datadog-logo-64x64.png</iconUrl>
```

File: src/Datadog.Trace/Datadog.Trace.nuspec [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>Manual instrumentation library for Datadog APM</description>
+    <releaseNotes>See release notes at https://github.com/DataDog/dd-trace-dotnet/releases</releaseNotes>
+    <copyright>Datadog, Inc. 2017-2019</copyright>
+    <tags>Datadog;APM;tracing;profiling;instrumentation</tags>
+    <repository type="git" url="https://github.com/DataDog/dd-trace-dotnet.git" branch="master" />
+    <dependencies>
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Datadog.Trace.nuspec]: We need to add framework dependencies to `System.Configuration` and `System.Web` when targeting `net45` and `net461`.

File: src/Datadog.Trace/Datadog.Trace.nuspec [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+      <dependency id="Microsoft.CSharp" version="4.5.0" />
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Datadog.Trace.nuspec]: I think `Microsoft.CSharp` is a development dependency only, so we can probably remove it here.

File: src/Datadog.Trace/Datadog.Trace.nuspec [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+      <dependency id="MsgPack.Cli" version="1.0.1" />
+      <dependency id="Newtonsoft.Json" version="12.0.1" />
+      <dependency id="System.Net.Http" version="4.3.4" />
+      <dependency id="System.Runtime.InteropServices.RuntimeInformation" version="4.3.0" />
+    </dependencies>
+    <summary>The core library for Datadog APM. This library is used for automatic tracing, manual tracing, and open tracing.</summary>
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Datadog.Trace.nuspec]: `<summary>` is deprecated in favor of `<description>`

```suggestion
    <description>The core library for Datadog APM. This library is used for automatic tracing, manual tracing, and open tracing.</description>
```

File: src/Datadog.Trace/Datadog.Trace.nuspec (hunk 14 above)
Comment (lucaspimentel) [inline src/Datadog.Trace/Datadog.Trace.nuspec]: We need to split `<dependencies>` into the three frameworks we target:

```xml
<dependencies>
  <group targetFramework=".NETFramework4.5">
	<dependency ... />
  </group>
  <group targetFramework=".NETFramework4.6.1">
	<dependency ... />
  </group>
  <group targetFramework=".NETStandard2.0">
	<dependency ... />
  </group>
</dependencies>
```

File: src/Datadog.Trace/Logging/DatadogLogging.cs [hunk 17]
```diff
@@ -0,0 +1,117 @@
...
+
+                var debugEnabledVariable = Environment.GetEnvironmentVariable("DD_TRACE_DEBUG")?.ToLower();
+                if (debugEnabledVariable != "1" && debugEnabledVariable != "true")
+                {
+                    // No verbose or debug logs
+                    MinimumLogEventLevel = LogEventLevel.Information;
+                }
+
+                var maxLogSizeVar = Environment.GetEnvironmentVariable("DD_MAX_LOGFILE_SIZE");
+                if (!string.IsNullOrEmpty(maxLogSizeVar) && long.TryParse(maxLogSizeVar, out var maxLogSize))
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Logging/DatadogLogging.cs]: No need for the extra check.

```suggestion
                if (long.TryParse(maxLogSizeVar, out var maxLogSize))
```

File: src/vendors/Datadog.Trace.Vendoring/DatadogLogging.cs (hunk 10 above)
Comment (lucaspimentel) [reply src/vendors/Datadog.Trace.Vendoring/DatadogLogging.cs]: Indeed, my suggestion doesn't even compile! 🤦‍♂

File: src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj (hunk 6 above)
Comment (lucaspimentel) [reply src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj]: > I figure we should do everything they do.

That argument won't work when we start adding more vendor files into this project. When are we going to two vendors to things differently? 😉

File: src/Datadog.Trace/Logging/DatadogLogging.cs [hunk 18]
```diff
@@ -0,0 +1,117 @@
...
+                        logDirectory = NixDefaultDirectory;
+                    }
+                    else
+                    {
+                        logDirectory = Environment.CurrentDirectory;
+                    }
+                }
+
+                // Ends in a dash because of the date postfix
+                var managedLogPath = Path.Combine(logDirectory, $"dotnet-tracer-{currentProcess.ProcessName}-.log");
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Logging/DatadogLogging.cs]: Looks like an extra `-` at the end.
```suggestion
                var managedLogPath = Path.Combine(logDirectory, $"dotnet-tracer-{currentProcess.ProcessName}.log");
```

File: src/Datadog.Trace/Logging/DatadogLogging.cs [hunk 19]
```diff
@@ -0,0 +1,117 @@
...
+namespace Datadog.Trace.Logging
+{
+    internal static class DatadogLogging
+    {
+        private const string NixDefaultDirectory = "/var/log/datadog/";
+        private static readonly string WindowsDefaultDirectory =
+            Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Datadog .NET Tracer", "logs");
+
+        private static readonly long? MaxLogFileSize = 10 * 1024 * 1024;
+        private static readonly LogEventLevel MinimumLogEventLevel = LogEventLevel.Verbose; // Lowest level
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Logging/DatadogLogging.cs]: This should be configurable, but we can leave that for a separate PR.

File: src/vendors/Datadog.Trace.Vendoring/DatadogLogging.cs (hunk 11 above)
Comment (lucaspimentel) [reply src/vendors/Datadog.Trace.Vendoring/DatadogLogging.cs]: Oops, never mind. I see now that you're setting this in the static ctor.

File: src/Datadog.Trace/Datadog.Trace.nuspec (hunk 13 above)
Comment (zacharycmontoya) [reply src/Datadog.Trace/Datadog.Trace.nuspec]: +1 for fixing the deprecated `iconUrl` at a later point

File: src/Datadog.Trace/Datadog.Trace.nuspec (hunk 14 above)
Comment (zacharycmontoya) [reply src/Datadog.Trace/Datadog.Trace.nuspec]: Our 1.7.0 auto-magically created NuGet package uses the following

```xml
    <frameworkAssemblies>
      <frameworkAssembly assemblyName="System.Configuration" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
      <frameworkAssembly assemblyName="System.Web" targetFramework=".NETFramework4.5, .NETFramework4.6.1" />
    </frameworkAssemblies>
```

File: src/Datadog.Trace/Logging/DatadogLogging.cs [hunk 20]
```diff
@@ -0,0 +1,117 @@
...
+                    {
+                        logDirectory = WindowsDefaultDirectory;
+                    }
+                    else if (Directory.Exists(NixDefaultDirectory))
+                    {
+                        logDirectory = NixDefaultDirectory;
+                    }
+                    else
+                    {
+                        logDirectory = Environment.CurrentDirectory;
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Logging/DatadogLogging.cs]: Should this fallback be something like `Path.GetTempPath()`? Users shouldn't generally have permission to write to the directory where applications are installed (like `C:\Program Files` on Windows). Web apps in IIS will also generally not be able to write to the directory they are running from.

File: src/Datadog.Trace/Datadog.Trace.nuspec (hunk 15 above)
Comment (zacharycmontoya) [reply src/Datadog.Trace/Datadog.Trace.nuspec]: 1.7.0 has the following, in case we decide to keep the dependency, and each of them have the same `exclude`.

```xml
        <dependency id="Microsoft.CSharp" version="4.5.0" exclude="Build,Analyzers" />
```

File: src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj (hunk 8 above)
Comment (zacharycmontoya) [reply src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj]: This seems like something that could be grouped into another set of changes, but shouldn't removing the `<LangVersion>` property work?

File: src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj [hunk 21]
```diff
@@ -0,0 +1,61 @@
...
+    <TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">net45;net461;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netstandard2.0</TargetFrameworks>
+    <GenerateDocumentationFile>false</GenerateDocumentationFile>
+    <LangVersion>7.3</LangVersion>
+
+    <!-- Strong name signature -->
+    <SignAssembly>true</SignAssembly>
+    <AssemblyOriginatorKeyFile>..\..\..\Datadog.Trace.snk</AssemblyOriginatorKeyFile>
+    <PublicSign Condition=" '$(OS)' != 'Windows_NT' ">true</PublicSign>
+    <DelaySign>false</DelaySign>
```
Comment (lucaspimentel) [inline src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj]: I don't think we need these since they're inherited from the root `Directory.Build.props`.

Comment (lucaspimentel) [reply src/vendors/Datadog.Trace.Vendoring/Datadog.Trace.Vendoring.csproj]: Never mind. I finally got the the end and noticed that you are intentionally blocking the parent `Directory.Build.props`.

File: tools/SynchronizeVersions/Program.cs [hunk 22]
```diff
@@ -65,6 +65,10 @@ public static void Main(string[] args)
                 "src/Datadog.Trace/Datadog.Trace.csproj",
```
Comment (zacharycmontoya) [inline tools/SynchronizeVersions/Program.cs]: I think this entry can be removed now, right?

File: src/Datadog.Trace/Logging/DatadogLogging.cs (hunk 18 above)
Comment (colin-higgins) [reply src/Datadog.Trace/Logging/DatadogLogging.cs]: it's not an extra dash, the comment above explains it.

File: src/Datadog.Trace/Logging/DatadogLogging.cs (hunk 19 above)
Comment (colin-higgins) [reply src/Datadog.Trace/Logging/DatadogLogging.cs]: It is configurable, this is just the default if it's NOT configured.

File: src/Datadog.Trace/Logging/DatadogLogging.cs (hunk 18 above)
Comment (lucaspimentel) [reply src/Datadog.Trace/Logging/DatadogLogging.cs]: Oops, reading comprehension fail. 🤦‍♂

File: src/Datadog.Trace/Logging/DatadogLogging.cs (hunk 19 above)
Comment (lucaspimentel) [reply src/Datadog.Trace/Logging/DatadogLogging.cs]: Never mind, then. Thanks!

File: src/Datadog.Trace/Datadog.Trace.csproj [hunk 23]
```diff
@@ -30,4 +30,31 @@
     <PackageReference Include="System.Runtime.InteropServices.RuntimeInformation" Version="4.3.0" />
   </ItemGroup>
 
+  <!-- Below this point is reserved for vendor items -->
+  <!--Serilog-->
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'net45' ">
+    <DefineConstants>REMOTING;HASHTABLE</DefineConstants>
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Datadog.Trace.csproj]: We should append to any previously defined constants instead of overwriting them:
```suggestion
    <DefineConstants>$(DefineConstants);REMOTING;HASHTABLE</DefineConstants>
```

Comment (lucaspimentel) [inline src/Datadog.Trace/Datadog.Trace.csproj]: We should append to any previously defined constants instead of overwriting them:
```suggestion
    <DefineConstants>$(DefineConstants);REMOTING;HASHTABLE</DefineConstants>
```

File: src/Datadog.Trace/Datadog.Trace.csproj [hunk 24]
```diff
@@ -30,4 +30,31 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'net461' ">
+    <DefineConstants>ASYNCLOCAL;HASHTABLE</DefineConstants>
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Datadog.Trace.csproj]:
```suggestion
    <DefineConstants>$(DefineConstants);ASYNCLOCAL;HASHTABLE</DefineConstants>
```

File: src/Datadog.Trace/Datadog.Trace.csproj [hunk 25]
```diff
@@ -30,4 +30,31 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
+    <DefineConstants>ASYNCLOCAL;HASHTABLE</DefineConstants>
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Datadog.Trace.csproj]:
```suggestion
    <DefineConstants>$(DefineConstants);ASYNCLOCAL;HASHTABLE</DefineConstants>
```

File: src/Datadog.Trace/Datadog.Trace.csproj [hunk 26]
```diff
@@ -30,4 +30,31 @@
...
+  </PropertyGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
+    <PackageReference Include="System.Collections.NonGeneric" Version="4.3.0" />
+  </ItemGroup>
+
+  <!--Serilog.Sinks.File-->
+  <PropertyGroup Condition="$(TargetFramework.StartsWith('net4'))">
+    <DefineConstants>ATOMIC_APPEND;HRESULTS</DefineConstants>
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Datadog.Trace.csproj]:
```suggestion
    <DefineConstants>$(DefineConstants);ATOMIC_APPEND;HRESULTS</DefineConstants>
```

File: src/Datadog.Trace/Datadog.Trace.csproj [hunk 27]
```diff
@@ -30,4 +30,31 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
+    <DefineConstants>OS_MUTEX</DefineConstants>
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Datadog.Trace.csproj]:
```suggestion
    <DefineConstants>$(DefineConstants);OS_MUTEX</DefineConstants>
```
